main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -1028,4 +1028,25 @@ object WeederError {
          |""".stripMargin
     }
   }
+
+
+  /**
+    * An error raised to indicate expression nesting beyond the threshold.
+    *
+    * @param hint     Optional hint with more details about the error
+    * @param loc      The source location.
+    */
+  case class TooNested(hint: Option[String] = None, loc: SourceLocation) extends WeederError with Recoverable {
+    def summary: String = {
+      s"Expression is too nested."
+    }
+
+    def message(fmt: Formatter): String = {
+      val hintStr = hint.map(s"\nHint: " + _).getOrElse("")
+      s""">> Expression is too nested.
+         |
+         |${fmt.code(loc, s"Here")}$hintStr
```

Comment:
Instead of the hint? 

I do still think that the issue of pasting of long chains of `::` is important so I wanted to have the error point them to `List#{}`.

Reply:
You can have that, but then we need two error classes. (See above).